### PR TITLE
docs(examples): add an example for streaming events from a watch endpoint

### LINKED ISSUE #219: `/watch` / streaming API example
Add an example to examples/ demonstrating calling a `/watch` endpoint, and setting up the Node.js streams.

### PR DESCRIPTION
**Pull request by silasbw**
Closes #219

### CODE REVIEW

**Comment by silasbw** [comment]
@srossross-tableau or @XVincentX is this a helpful example?

**Comment by srossross-tableau** [comment]
Yes, thanks

**Comment by XVincentX** [comment]
Cool! I'll have a look into!

File: examples/watch.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+const Client = require('../lib').Client;
+const config = require('../lib').config;
+const JSONStream = require('json-stream');
+
+const deploymentManifest = require('./nginx-deployment.json');
+
+async function cleanup(client) {
+  try {
+    await client.apis.apps.v1.namespaces('default').deployments(deploymentManifest.metadata.name).delete();
+  } catch (err) { }
```
Comment (XVincentX) [inline examples/watch.js]: You might want to `console.error` at least.

File: examples/watch.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+async function triggerEvents(client) {
+  //
+  // Trigger some events
+  //
+  await cleanup(client);
```
Comment (XVincentX) [inline examples/watch.js]: I think this line should be outside of this function.

File: examples/watch.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  for (let count = 0; count < 3; count++) {
+    await new Promise(resolve => setTimeout(resolve, 1000));
```
Comment (XVincentX) [inline examples/watch.js]: Why do you need to wait here?

File: examples/watch.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    await client.apis.apps.v1.namespaces('default').deployments.post({ body: deploymentManifest });
+    await new Promise(resolve => setTimeout(resolve, 1000));
```
Comment (XVincentX) [inline examples/watch.js]: ditto

File: examples/watch.js (hunk 3 above)
Comment (silasbw) [reply examples/watch.js]: I want to delay 1 second before executing the next line ("`sleep 1`")

File: examples/watch.js (hunk 4 above)
Comment (silasbw) [reply examples/watch.js]: Same :)

**Comment by silasbw** [comment]
Thanks for the feedback @XVincentX and sorry for the delay in applying it.  If you have time for another look, i'd appreciate it.

**Pull request merged**